client/src/main/java/com/aws/greengrass/cli/util/logs/impl/AggregationImpl.java

```diff
@@ -49,36 +52,39 @@ public void configure(boolean follow, Filter filter, int before, int after, int
     /*
      * Read log files from input commands.
      *
-     * @param logFileArray an array of file paths
-     * @param logDirArray an array of file directories
+     * @param logFileList a list of file paths
+     * @param logDirList a list of file directories
      * @return a PriorityBlockingQueue containing log entries. Log entries from multiple files
      *  will read into this queue concurrently ordered by their timestamps.
      */
     @Override
-    public LogQueue readLog(String[] logFileArray, String[] logDirArray) {
+    public LogQueue readLog(List<Path> logFileList, List<Path> logDirList) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.